nearcache.go

```diff
@@ -0,0 +1,692 @@
+		value:          av,
+		ExpirationTime: expirationTime,
+	}
+}
+
+func (r *nearCacheRecord) Value() interface{} {
+	return r.value.Load()
+}
+
+func (r *nearCacheRecord) SetValue(value interface{}) {
+	r.value.Store(&value)
+}
+
+func (r *nearCacheRecord) Hits() int32 {
+	return atomic.LoadInt32(&r.hits)
+}
+
+func (r *nearCacheRecord) SetHits(value int32) {
+	atomic.StoreInt32(&r.hits, value)
+}
+
+func (r *nearCacheRecord) IncrementHits() {
+	atomic.AddInt32(&r.hits, 1)
+}
+
+func (r *nearCacheRecord) ReservationID() int64 {
+	return atomic.LoadInt64(&r.reservationID)
+}
+
+func (r *nearCacheRecord) SetReservationID(rid int64) {
+	atomic.StoreInt64(&r.reservationID, rid)
+}
+
+func (r *nearCacheRecord) LastAccessTimeMS() int64 {
+	return atomic.LoadInt64(&r.lastAccessTime)
+}
+
+func (r *nearCacheRecord) SetLastAccessTimeMS(ms int64) {
+	atomic.StoreInt64(&r.lastAccessTime, ms)
+}
+
+func (r *nearCacheRecord) IsExpiredAtMS(ms int64) bool {
+	return r.ExpirationTime > 0 && r.ExpirationTime <= ms
+}
+
+func (r *nearCacheRecord) IsIdleAtMS(maxIdleMS, nowMS int64) bool {
+	if maxIdleMS <= 0 {
+		return false
+	}
+	lat := r.LastAccessTimeMS()
+	if lat > 0 {
+		return lat+maxIdleMS < nowMS
+	}
+	return r.CreationTime+maxIdleMS < nowMS
+}
+
+func (r *nearCacheRecord) CachedAsNil() bool {
+	return atomic.LoadInt32(&r.cachedAsNil) == 1
+}
+
+func (r *nearCacheRecord) SetCachedAsNil() {
+	atomic.StoreInt32(&r.cachedAsNil, 1)
+}
+
+// nearCacheStats contains statistics for a Near Cache instance.
+type nearCacheStats struct {
+	invalidationRequests int64
+	// Misses is the number of times a key was not found in the Near Cache.
+	Misses int64
+	// Hits is the number of times a key was found in the Near Cache.
+	Hits int64
+	// Expirations is the number of expirations.
+	Expirations int64
+	// Evictions is the number of evictions.
+	Evictions int64
+	// OwnedEntryCount is the number of entries in the Near Cache.
+	OwnedEntryCount int64
+	// OwnedEntryMemoryCost is the estimated memory cost of the entries in the Near Cache.
+	OwnedEntryMemoryCost int64
+	// Invalidations is the number of successful invalidations.
+	Invalidations int64
+	// LastPersistenceKeyCount is the number of keys saved in the last persistence task.
+	LastPersistenceKeyCount int64
+	// LastPersistenceWrittenBytes is the size of the last persistence task.
+	LastPersistenceWrittenBytes int64
+	// PersistenceCount is the number of completed persistence tasks.
+	PersistenceCount int64
+	// CreationTime is the time the Near Cache was initialized.
+	CreationTime time.Time
+	// LastPersistenceTime is the time of the last completed persistence task.
+	LastPersistenceTime time.Time
+	// LastPersistenceFailure is the error message of the last completed persistence task.
+	LastPersistenceFailure string
+	// LastPersistenceDuration is the duration of the last completed persistence task.
+	LastPersistenceDuration time.Duration
+}
+
+func (st *nearCacheStats) InvalidationRequests() int64 {
```

Comment:
This is not used for now, right? Will it be used in later PRs?

Reply:
It seems not, removed at: https://github.com/hazelcast/hazelcast-go-client/pull/806/commits/aaa18f6bacb80c4aacf00ef00b53024843e53fbb